src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+import { Flex, PseudoBox, useTheme } from '@chakra-ui/core';
+
+const LEFT_PAGE = 'LEFT';
+const RIGHT_PAGE = 'RIGHT';
+
+/**
+ * @function Pagination
+ *
+ * @param {number} totalRecords - Total records count
+ * @param {function} onPageChanged - Callback function that passes paginated information
+ * @param {number} pageLimit - Number of items to display per page
+ * @param {number} pageNeighbors - Number of neighboring pages when pagination is in the middle of the list
+ *
+ */
+
+const Pagination = props => {
+  const theme = useTheme();
+
+  const [totalRecords, setTotalRecords] = useState(null);
+  const [totalPages, setTotalPages] = useState(0);
+  const [pageLimit, setPageLimit] = useState(10);
+  const [pageNeighbors, setPageNeighbors] = useState(1);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [pages, setPages] = useState([]);
+
+  useEffect(() => {
+    const { totalRecords, pageLimit, pageNeighbors } = props;
+
+    if (pageLimit) {
+      setPageLimit(pageLimit);
+    }
+    if (pageNeighbors) {
+      setPageNeighbors(Math.max(0, Math.min(pageNeighbors, 2)));
+    }
+
+    const totalPages = Math.ceil(totalRecords / pageLimit);
+    setTotalRecords(totalRecords);
+    setTotalPages(totalPages);
+  }, []);
+
+  useEffect(() => {
+    setPages(fetchPageNumbers());
```

Comment:
@billy-le This should definitely be addressed soon too. This is a major concern here.

`fetchPageNumbers()` should be declared within this `useEffect()` call so that you can properly declare it's dependencies which include `pageNeighbors`, `totalPages`, and `currentPage` among others.